components/scream/src/physics/p3/tests/p3_run_and_cmp.cpp

```diff
@@ -93,13 +93,19 @@ static Int compare (const std::string& label, const Scalar* a,
 }
 
 struct Baseline {
-  Baseline (const Int nsteps=6, const Int ncol=3, const Int repeat=0, const std::string predict_nc="both") :
+  Baseline (const Int nsteps=6, const Int ncol=3, const Int repeat=0, const std::string predict_nc="both", const std::string prescribed_ccn="both") :
     m_ic_ncol(ncol), m_repeat(repeat)
   {
-    for (const bool do_predict_nc : {true, false}) {
-      if (predict_nc == "both" || ( (do_predict_nc and predict_nc == "yes") || (!do_predict_nc and predict_nc == "no") )) {
-        //                 initial condit,     dt,  nsteps, prescribe or predict nc
-        params_.push_back({ic::Factory::mixed, 300, nsteps, do_predict_nc});
+    for (const bool do_prescribed_CCN : {true, false}) {
+      for (const bool do_predict_nc : {true, false}) {
+        if (predict_nc == "both" || ( (do_predict_nc and predict_nc == "yes") || (!do_predict_nc and predict_nc == "no") )) {
+          //                 initial condit,     dt,  nsteps, prescribe or predict nc
+          params_.push_back({ic::Factory::mixed, 300, nsteps, do_predict_nc});
```

Comment:
Doesn't the ParamSet struct have 5 elements now?  How does this line and the one below switch between whther `do_predict_nc` is being set or `do_prescribed_CCN`?

Reply:
I agree with Aaron - params_.push_back should contain both the do_predict_nc and do_prescribed_CCN state. This breaks the weird "loop over all options but use an if statement to decide whether or not to actually do anything" logic you've inherited here. I think fixing this will require rethinking how the logic works (which may be best to save for a follow-up PR).

I was initially very confused why the code doesn't crash when passed the wrong number of args. My recollection is that any entry to params_ which isn't assigned is given the value of 0. Thus you're only testing the "false" branch of the thing you don't specify.

Reply:
Yes, this is highly possibly the test code only test partial branch. The fix is committed.

Reply:
Thanks for putting the fix in xyuan.